kombu/utils/json.py

```diff
@@ -3,88 +3,66 @@
 from __future__ import annotations
 
 import base64
-import datetime
-import decimal
 import json
 import uuid
+from datetime import date, datetime, time
+from decimal import Decimal
+from typing import Any, Callable, Optional, TypeVar
 
-try:
-    from django.utils.functional import Promise as DjangoPromise
-except ImportError:  # pragma: no cover
-    class DjangoPromise:
-        """Dummy object."""
+textual_types = ()
 
+try:
+    from django.utils.functional import Promise
 
-class _DecodeError(Exception):
```

Comment:
I removed this _DecodeError, because it was not raised anywhere.

It was leftover for compatibility with `simplejson` library which is long gone.